packages/text-area/src/vaadin-text-area-mixin.js

```diff
@@ -209,6 +219,36 @@ export const TextAreaMixin = (superClass) =>
       }
     }
 
+    /** @private */
+    __updateMaxHeight(maxRows) {
```

Comment:
This is the best I could think of, though it has some flaws:
- Borders are ignored, though it seems unlikely that they would be used in inner parts of the component. We could include them if it becomes necessary.
- If `line-height` computes to a non-numeric value such as `normal`, then effectively no max height will be set. That seems to only be the case if `normal` is set explicitly.

I was also considering implementing some measurement similar to what is done in `_updateHeight`. That would involve temporarily setting `rows` to the max. number of rows and then measuring the resulting height of the input container, plus a lot of stuff around that to prepare for measurement and restoring some initial values after. That felt more complex and brittle overall.

But maybe I'm missing something crucial with this approach, feel free to suggest alternatives if you can think of some.